### ISSUE TITLE
acme_diags missing for OSX-64 uvcdat install

### ISSUE THREAD

**Issue by durack1** [UV-CDAT/uvcdat#2195]
```
machine:~ paul$ conda create -n uvcdat2120 uvcdat -c conda-forge -c uvcdat
Fetching package metadata .............
Solving package specifications: 

PackageNotFoundError: Dependency missing in current osx-64 channels: 
  - uvcdat -> acme_diags
```

**Comment by durack1** [comment]
@doutriaux1 ping!

**Comment by chengzhuzhang** [comment]
@charlesdoutriaux I'm just adding, Jerry had the same error.

**Comment by durack1** [comment]
@doutriaux1 @zshaheen it seems `acme_diags` is now available from the dedicated [acme](https://anaconda.org/acme/acme_diags/files) conda channel - when did this change? The existing environment `*.yml` files will need to be updated at https://github.com/UV-CDAT/uvcdat/wiki/Install

**Comment by durack1** [comment]
@doutriaux1 @zshaheen this is fixable by adding the new channel:
```
vpnb-user-128-15-246-252:~ paul$ conda create -n uvcdat2120 uvcdat -c conda-forge -c uvcdat -c acme
Fetching package metadata ...............
Solving package specifications: .

Package plan for installation in environment /Users/paul/anaconda2/envs/uvcdat2120:

The following NEW packages will be INSTALLED:

    acme_diags:                         1.0.1-h48f417c_0          acme       
    alabaster:                          0.7.10-py27_1             conda-forge
...
```
But I wonder whether adding yet another package to the default UV-CDAT install is a good idea?

**Comment by zshaheen** [comment]
We didn't want to manage `acme_diags` on two Anaconda channels. If someone did `conda install acme_diags -c conda-forge -c uvcdat -c acme`, they would get `acme_diags` from the `uvcdat` channel, and again we didn't want to manage two channels. So we removed the `acme_diags` package from the `uvcdat` channel.

@durack1 Yeah, we can just add the `acme` channel like how you did.

**Comment by chengzhuzhang** [comment]
@zshaheen @doutriaux1 
I'm not sure if it's a good idea to add acme_diags in uvcdat. Any suggestions?

**Comment by durack1** [comment]
@zshaheen my question was related to why `acme_diags` is a dependency for the UV-CDAT install. We haven't added the `pcmdi_metrics` as a dependency, rather we add `uvcdat` as a dependency of the PMP. I'd recommend that we reduce the UV-CDAT dependencies as much as possible, so that when you issue
```
> conda create -n uvcdat2120 -c conda-forge -c uvcdat uvcdat
```
You just get the bare essentials for the UV-CDAT installation, nothing extra or extraneous

**Comment by zshaheen** [comment]
@durack1 No idea. I think this was a precursor to the acme unified environment, which isn't really related to this anymore.

**Comment by chengzhuzhang** [comment]
I agree with @durack1. I think it makes more sense if we can follow the pcmdi_metrics' practice. Similarly, acme_diags is based on uvcdat and not a dependency of uvcdat. It won't function, without proper configuration and supplied datasets.

**Comment by durack1** [comment]
@doutriaux1 @zshaheen currently we have **218!!** package dependencies for the uvcdat installation, this is kinda nuts and there must be a bunch of opportunities to reduce this. Keep it simple I think!

[171206_UVCDATCondaInstall.txt](https://github.com/UV-CDAT/uvcdat/files/1536317/171206_UVCDATCondaInstall.txt)

**Comment by durack1** [comment]
@zshaheen regarding your https://github.com/UV-CDAT/uvcdat/issues/2195#issuecomment-349720399 above
> We didn't want to manage acme_diags on two Anaconda channels. If someone did conda install acme_diags -c conda-forge -c uvcdat -c acme, they would get acme_diags from the uvcdat channel, and again we didn't want to manage two channels. So we removed the acme_diags package from the uvcdat channel.

Wouldn't this be fixed by just reordering your channels:
```
conda install acme_diags -c acme -c conda-forge -c uvcdat
```
Or conversely, following my "not a dependency" suggestion you could just do the single line (which then allows a cleanup on the UV-CDAT dependency list):
```
conda create -n acmeDiagsEnv -c acme -c conda-forge -c uvcdat uvcdat acme_diags
```

**Comment by zshaheen** [comment]
Yes you're right. However, some of our users mix up the order, for whatever reasons. I think we should just remove `acme_diags`, among other packages, from uvcdat.

**Comment by doutriaux1** [comment]
@durack1 I took `acme_diags` out of uvcdat, please try again and let us know.

**Comment by durack1** [comment]
@zshaheen regarding
> Yes you're right. However, some of our users mix up the order, for whatever reasons. I think we should just remove acme_diags, among other packages, from uvcdat.

I'm looking at @potter2 here, I don't think we should add dependencies to resolve sloppy conda command line calls, rather let's just educate folks to do the right thing. I've personally been sloppy with my conda channel ordering in the past, and have just educated myself to fix my own problem.

Let's try and keep the UV-CDAT package and install/conda stuff as lean and clean as possible, so that we're not constantly tripping over ourselves, like in this issue.

**Comment by durack1** [comment]
@doutriaux1 great, fixed:
```
machine:~ paul$ conda create -n uvcdat2120 -c conda-forge -c uvcdat uvcdat
Fetching package metadata .............
Solving package specifications: .

Package plan for installation in environment /Users/paul/anaconda2/envs/uvcdat2120:

The following NEW packages will be INSTALLED:

    alabaster:                          0.7.10-py27_1             conda-forge
...
```

I'm not sure if the "clean up dependencies" comments scattered above should warrant a new issue - not sure how to catch this, as the original issue is now resolved

**Comment by durack1** [comment]
@doutriaux1 the files:
https://raw.githubusercontent.com/UV-CDAT/uvcdat/master/conda/2.12.yml
https://raw.githubusercontent.com/UV-CDAT/uvcdat/master/conda/2.12-osx.yml
https://raw.githubusercontent.com/UV-CDAT/uvcdat/master/conda/2.12-nox.yml
https://raw.githubusercontent.com/UV-CDAT/uvcdat/master/conda/2.12-nox-osx.yml

Will all need updating - these are linked off https://github.com/UV-CDAT/uvcdat/wiki/Install

**Comment by zshaheen** [comment]
@doutriaux1 @durack1 Done in c0dcf2fac2d41f7140a81bf9233c30287ccc6785

**Comment by durack1** [comment]
@zshaheen thanks, almost there, the `conda-forge/label/broken` channel needs to also be added to resolve the issue:
```
machine:~ paul$ conda env create -f ~/Desktop/2.12-nox-osx.yml
Using Anaconda API: https://api.anaconda.org
Fetching package metadata .............

NoPackagesFoundError: Package missing in current osx-64 channels: 
  - esmf 7.0.0 6
```

**Comment by durack1** [comment]
@zshaheen nope, I changed the top channel info:
```
name: 2.12-nox
channels:
- conda-forge
- uvcdat
- defaults
- conda-forge/label/broken
...
```
And now I hit:
```
machine:~ paul$ conda env create -f ~/Desktop/2.12-nox-osx.yml
Using Anaconda API: https://api.anaconda.org
Fetching package metadata ...............

NoPackagesFoundError: Package missing in current osx-64 channels: 
  - uvcdat 2.12 h5103ec1_0
```

**Comment by durack1** [comment]
Sheez, I just updated (again) - after look at https://anaconda.org/uvcdat/uvcdat/files
```
name: 2.12-nox
channels:
- conda-forge
- uvcdat
- defaults
- conda-forge/label/broken
- uvcdat/label/v2.12_0 ! WHY WAS THIS DONE??
...
```
And now get
```
vpnb-user-128-15-246-252:~ paul$ conda env create -f ~/Desktop/2.12-nox-osx.yml
Using Anaconda API: https://api.anaconda.org
Fetching package metadata .................
Solving package specifications: 

NoPackagesFoundError: Dependency missing in current osx-64 channels: 
  - uvcdat 2.12 h5103ec1_0 -> acme_diags
```
So I give up

**Comment by durack1** [comment]
Ok, so the attached now works (by removing the hash part of the uvcdat version)
[2.12-nox-osx.yml.txt](https://github.com/UV-CDAT/uvcdat/files/1536672/2.12-nox-osx.yml.txt)

**Comment by durack1** [comment]
This issue was fixed by #2196, however not sure how best to trim back down the expansive dependency lists..
